test/SlowTests/Issues/RavenDB-21399-2.cs

```diff
@@ -133,4 +140,87 @@ public void CanSafelySwapLeavesWithoutRemovingTermFromDisk()
             tree.TryRemove($"{--k}", out _);
         }
     }
+
+   [RavenTheory(RavenTestCategory.Voron | RavenTestCategory.Corax)]
+   [InlineData(1274580600)]
+    public void RemoveBranchAndMoveAllItemsIntoAnother(int seed)
+    {
+        using var wtx = Env.WriteTransaction();
+        CompactTree tree = wtx.CompactTreeFor($"test");
+        var lookup = tree._inner;
+        long k = 0;
+        while (lookup.State.BranchPages < 4)
+        {
+            tree.Add($"{k}", k++);
+        }
+
+        var list = Enumerable.Range(0, (int)k).ToList();
+        var random = new Random(seed);
+        
+        while (list.Count > 0)
+        {
+            var index = random.Next(0, list.Count);
+            var elementToRemove = list[index];
+            Assert.True(tree.TryRemove($"{elementToRemove}", out _));
+            list.RemoveAt(index);
+        }
+    }
+    
+    [RavenTheory(RavenTestCategory.Voron | RavenTestCategory.Corax)]
+    [InlineData(228422771, 2000, 5_834_460)]
+    public void TermWillNotHitMaximumReferenceCount(int seed, int maximumElements, long operationCount)
```

Comment:
I would add a few other failure cases in order to cover other potential cases (if they exist)